File: JSLib/src/MsgRoute/MsgRouteBase.js
namespace SDK.Lib
{
    public enum MsgRouteType
    {
        eMRT_BASIC,      // 基本类型
    }

    public enum MsgRouteID
    {
        eMRIDSocketOpened,      // socket Opened
        eMRIDSocketClosed,      // socket Closed
        eMRIDLoadedWebRes,      // web 资源加载完成
        eMRIDThreadLog,      // 线程打日志
    }

    public class MsgRouteBase : IRecycle, IDispatchObject
    {
        public MsgRouteType m_msgType;
        public MsgRouteID m_msgID;          // 只需要一个 ID 就行了

        public MsgRouteBase(MsgRouteID id)
        {
            m_msgType = MsgRouteType.eMRT_BASIC;
            m_msgID = id;
        }

        virtual public void resetDefault()
        {

        }
    }
}

File: PyLibs/src/Libs/MsgRoute/MsgRouteBase.py
# -*- coding: utf-8 -*-

from Libs.Core.GObject import GObject;

class MsgRouteType(GObject):
    eMRT_BASIC = 0;      # 基本类型


class MsgRouteID(GObject):
    eMRIDSocketOpened = 0;      # socket Opened
    eMRIDSocketClosed = 1;      # socket Opened
    eMRIDLoadedWebRes = 2;      # web 资源加载完成
    eMRIDThreadLog = 3;         # 线程打日志


class MsgRouteBase(GObject):
    
    def __init__(self, uniqueId):
        super(GObject, self).__init__();
        
        self.mTypeId = "MsgRouteBase";

        self.m_msgType = MsgRouteType.eMRT_BASIC;
        self.m_msgID = uniqueId;


    def resetDefault(self):
        pass;

